pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -552,17 +554,23 @@ func (a *Actuator) provisionHost(ctx context.Context, host *bmh.BareMetalHost,
 	config *bmv1alpha1.BareMetalMachineProviderSpec) error {
 	originalHost := host.DeepCopy()
 
-	// We only want to update the image setting if the host does not
```

Comment:
Changing this is potentially breaking as it will cause re-deploys on upgrade. Maybe skip everything if `host.Spec.Image != nil || host.Spec.CustomDeploy != nil`?

Reply:
This comment was at best out of date and at worst nonsense. Kiran was right to delete it.
Hosts that are already provisioned/ing are not going to be selected anyway, so we will never get to this point.